### PR TITLE
Openflow on bmv2 support.

### PR DESCRIPTION
**Pull request by pierce-m**

### CODE REVIEW

File: tests/of-tests/openflow.py.in [hunk 1]
```diff
@@ -420,61 +417,6 @@ def runTest(self):
         self.controller.message_send(req)
         do_barrier(self.controller)
 
-class TableStatsGet(openflow_base_tests.OFTestInterface):
```
Comment (pierce-m) [inline tests/of-tests/openflow.py.in]: I'm getting rid of this test since global table counters are not supported in bmv2.

File: .travis.yml [hunk 2]
```diff
@@ -80,30 +90,42 @@ install:
...
   - sleep 10
-  - sudo PYTHONPATH=$PYP_ADD:$PYTHONPATH ./bmv2/run_tests.sh --test-dir tests/ptf-tests/api-tests/
+  - $PD_TEST_SCRIPT
   - sudo killall bmswitchp4_drivers lt-bmswitchp4_drivers simple_switch lt-simple_switch; echo 0
   - sudo ./bmv2/run_bm.sh &>/dev/null &
   - sleep 10
   - sudo ./bmv2/run_drivers.sh &>/dev/null &
   - sleep 10
-  - sudo PYTHONPATH=$PYP_ADD:$PYTHONPATH ./bmv2/run_tests.sh --test-dir tests/ptf-tests/sai-tests
+  - $OF_TEST_SCRIPT
```
Comment (antoninbas) [inline .travis.yml]: is there a reason why you did not move the testing code for PD+API+SAI and the testing code for OF to separate scripts under `CI/travis`, e.g. `run_switch_test.sh` and `run_of_test.sh`. Then this whole Travis file becomes much simpler.
Something like this:
```
matrix:
  - TEST_SCENARIO="switch"
  - TEST_SCENARIO="of"
...
script:
  - if [ "$TEST_SCENARIO" != "switch" ]; then ./configure --with-bmv2 --with-switchsai; fi
  - if [ "$TEST_SCENARIO" != "of" ]; then ./configure --with-bmv2 --with-of; fi
  - if [ "$TEST_SCENARIO" != "switch" ]; then ./CI/travis/run_switch_test.sh; fi
  - if [ "$TEST_SCENARIO" != "of" ]; then ./CI/travis/run_of_test.sh; fi
```
It seems that with the current file, you sometimes start the switch & drivers (+ sleep) for nothing.

File: .travis.yml [hunk 3]
```diff
@@ -12,7 +12,17 @@ services:
   - docker
 
 env:
-  - PYP_ADD="/usr/local/lib/python2.7/dist-packages:/usr/local/lib/python2.7/site-packages"
+  global:
+    - PYP_ADD="/usr/local/lib/python2.7/dist-packages:/usr/local/lib/python2.7/site-packages"
+    - KILL_BMV2="killall bmswitchp4_drivers lt-bmswitchp4_drivers simple_switch lt-simple_switch; echo 0"
```
Comment (antoninbas) [inline .travis.yml]: I don't think this is used

**Comment by antoninbas** [comment]
everything looks good save for the Travis file

**Comment by antoninbas** [comment]
+1

**Comment by antoninbas** [comment]
remember to squash when merging

**Pull request merged**